src/client/mod.rs

```diff
@@ -159,6 +159,34 @@ impl<P: Provider> Client<P> {
         let token = P::Token::from_response(&json)?;
         Ok(token)
     }
+
+    /// Requests an access token using an authorization code.
+    ///
+    /// See [RFC 6749, section 4.1.3](http://tools.ietf.org/html/rfc6749#section-4.1.3).
+    pub fn request_token_with<I, K, V>(
```

Comment:
`request_token` should be implemented by calling this with empty `pairs` to avoid duplication.